lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -37,237 +39,1042 @@
+          vrf_extension_template: Default_VRF_Extension_Universal
+          source: null
+          service_vrf_template: null 
+        - vrf_name: ansible-vrf-r2
+          vrf_id: 9008012
+          vrf_template: Default_VRF_Universal
+          vrf_extension_template: Default_VRF_Extension_Universal
+          source: null
+          service_vrf_template: null
+          
+- name: Delete all the VRFs
+    dcnm_vrf:
+      fabric: vxlan-fabric
+      state: deleted
+      
+- name: Create VRFs
+    dcnm_vrf:
+      fabric: vxlan-fabric
+      state: query
+      config:
+        - vrf_name: ansible-vrf-r1
+          vrf_id: 9008011
+          vrf_template: Default_VRF_Universal
+          vrf_extension_template: Default_VRF_Extension_Universal
+          source: null
+          service_vrf_template: null
+        - vrf_name: ansible-vrf-r2
+          vrf_id: 9008012
+          vrf_template: Default_VRF_Universal
+          vrf_extension_template: Default_VRF_Extension_Universal
+          source: null
+          service_vrf_template: null
 '''
 
+import datetime
+def logit(msg):
+    with open('/tmp/alog.txt', 'a') as of:
+        d = datetime.datetime.now().replace(microsecond=0).isoformat()
+        of.write("---- %s ----\n%s\n" % (d,msg))
 
-def check_vrf_exists(module):
 
-    fabric = module.params['fabric']
-    vrf = module.params['vrf_name']
-    path = '/rest/top-down/fabrics/{}/vrfs/{}'.format(fabric, vrf)
+class DcnmVrf:
 
-    response = dcnm_send(module, 'GET', path)
+    def __init__(self, module):
+        self.module = module
+        self.params = module.params
+        self.fabric = module.params['fabric']
+        self.config = module.params.get('config')
+        self.check_mode = False
+        self.conn = Connection(module._socket_path)
+        self.have_create = []
+        self.want_create = []
+        self.diff_create = []
+        self.have_attach = []
+        self.want_attach = []
+        self.diff_attach = []
+        self.have_deploy = {}
+        self.want_deploy = {}
+        self.diff_deploy = {}
+        self.diff_delete = {}
+        self.query = []
+        self.ip_sn = get_fabric_inventory_details(self.module, self.fabric)
 
-    if isinstance(response, dict):
-        if response.get('vrfName') == vrf:
-            return True
 
-    if isinstance(response, list):
-        if response and response[0]:
-            if response[0].get('ERROR') == 'Bad Request':
-                return False
+    def diff_for_attach_deploy(self, want_a, have_a):
+
+        attach_list = list()
+
+        if not want_a:
+            return attach_list
+
+        dep_vrf = False
+        for want in want_a:
+            found = False
+            if have_a:
+                for have in have_a:
+                    if want['serialNumber'] == have['serialNumber']:
+                        found = True
+
+                        ## Is this allowed use case? User wants to update vlan on an already attached and deployed VRF?
```

Comment:
@mikewiebe - Just verified this behaviour - updating a vlan on an already deployed VRF does not need detach/re-deploy. Just deploy would suffice.